presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/db/DbFunctionNamespaceManagerConfig.java

```diff
@@ -0,0 +1,171 @@
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class DbFunctionNamespaceManagerConfig
+{
+    private static final Splitter NAMESPACE_SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+    private static final Splitter CATALOG_SCHEMA_SPLITTER = Splitter.on('.');
+
+    private String functionNamespacesTableName = "sql_invoked_function_namespaces";
+    private String functionsTableName = "sql_invoked_functions";
+    private List<CatalogSchemaName> defaultFunctionNamespaces = ImmutableList.of();
+    private Duration queryTimeout = new Duration(20, SECONDS);
+
+    private int maxCatalogNameLength = 128;
+    private int maxSchemaNameLength = 128;
+    private int maxFunctionNameLength = 256;
+    private int maxParameterTypesLength = 500;
+    private int maxReturnTypeLength = 256;
+
+    @NotNull
+    public String getFunctionNamespacesTableName()
+    {
+        return functionNamespacesTableName;
+    }
+
+    @Config("function-namespace-table-name")
+    public DbFunctionNamespaceManagerConfig setFunctionNamespacesTableName(String functionNamespacesTableName)
+    {
+        this.functionNamespacesTableName = functionNamespacesTableName;
+        return this;
+    }
+
+    @NotNull
+    public List<CatalogSchemaName> getDefaultFunctionNamespaces()
+    {
+        return defaultFunctionNamespaces;
+    }
+
+    @Config("default-function-namespaces")
+    public DbFunctionNamespaceManagerConfig setDefaultFunctionNamespaces(String defaultFunctionNamespaces)
+    {
+        this.defaultFunctionNamespaces = NAMESPACE_SPLITTER.splitToList(defaultFunctionNamespaces).stream()
+                .map(s -> CATALOG_SCHEMA_SPLITTER.splitToList(s))
+                .map(parts -> {
+                    checkArgument(parts.size() == 2, format("Expect catalog.schema, get %s", Joiner.on(',').join(parts)));
+                    return new CatalogSchemaName(parts.get(0), parts.get(1));
+                })
+                .collect(toImmutableList());
+        return this;
+    }
+
+    @NotNull
+    public String getFunctionsTableName()
+    {
+        return functionsTableName;
+    }
+
+    @Config("function-table-name")
+    public DbFunctionNamespaceManagerConfig setFunctionsTableName(String functionsTableName)
+    {
+        this.functionsTableName = functionsTableName;
+        return this;
+    }
+
+    @NotNull
+    public Duration getQueryTimeout()
+    {
+        return queryTimeout;
+    }
+
+    @Config("query-timeout")
+    public DbFunctionNamespaceManagerConfig setQueryTimeout(Duration queryTimeout)
+    {
+        this.queryTimeout = queryTimeout;
+        return this;
+    }
+
+    @Min(1)
+    public int getMaxCatalogNameLength()
+    {
+        return maxCatalogNameLength;
+    }
+
+    @Config("max-catalog-name-length")
+    public DbFunctionNamespaceManagerConfig setMaxCatalogNameLength(int maxCatalogNameLength)
```

Comment:
You are not using these when creating the database tables. And as long as you are using `CREATE TABLE IF NOT EXISTS` you cannot really change the upper bound of this. So either restrict the allowed value, or this doesn't really need to be configurable.

Reply:
The Dao is very much MySQL syntax so the argument that people might keep that while using a different database backend is not very convincing to me. In general it's better to only provide flexibility when absolutely needed otherwise you might end up just creating dead code that hard to reason about later. I'd remove these until it's proven necessary. After all this is not hard to add when it's needed. Currently from this PR I don't see any use for this.